### DISCUSSION TITLE
get go and apiextensionsv1.JSON

### DISCUSSION THREAD

**Discussion by nyarly** [cue-lang/cue#1240] [Q&A]
I've got a `cue get go` generated file with an issue that I'm trying to wrangle. The source line I'm concerned with is

https://github.com/prometheus-operator/prometheus-operator/blob/c336e6348a6afc56fa1a5fc799efc3600b94d97d/pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go#L120

The resulting Cue omits that line entirely, which means I can't properly configure AlertmanagerConfigs with CUE based on the generated code. We're several steps deep into a structure, and the result is a def in a generated package.

So, I'm not sure how to proceed here? Edit the generated CUE?

**Comment by verdverm** [comment]
Is the Go source for the k8s package referenced available when you `cue get go`?

The `cue.mod/usr` dir is intended for augmenting the generated output, however since definitions are closed and the is a missing field, I don't think that will help in this case.

We should figure out the reason it is missing. Is it missing go deps during the import or a bug in the logic?

**Reply by nyarly** [reply]
Two things strike me: first, that (empirically) _all_ the code generated by `get go` produces defs, so `/usr` overrides aren't going to be terribly useful in a lot of cases.

Second, I am guessing that `get go` looked at `apiextensionsv1.JSON` and said "I don't know what this is, so I'm not putting anything." The comment in the original Go makes it clear: they're working around a limit in CRDs that prevents recursive types being validated. 

Maybe `routes: _` would've been better for `get go` to generate, since I'd be able to constrain from that in `/usr`, right?

**Reply by verdverm** [reply]
Yes this sounds about right. I would expect(ed behavior) CUE to put _ there rather than omitting the field.

Still the question is why is CUE not omitting the field at all...

**Reply by verdverm** [reply]
I'd probably add a debug print at the end of this function: https://github.com/cue-lang/cue/blob/master/cmd/cue/cmd/get_go.go#L913

Either way, this is the file of most interest.
